src/backend/cpu/kernel/regions.hpp

```diff
@@ -98,78 +99,77 @@ static void setUnion(LabelNode<T>* x, LabelNode<T>* y)
 template<typename T>
 void regions(Array<T> out, const Array<char> in, af_connectivity connectivity)
 {
-    const af::dim4 in_dims = in.dims();
-    const char *in_ptr  = in.get();
-    T    *out_ptr = out.get();
+    const af::dim4 inDims = in.dims();
+    const char *inPtr  = in.get();
+    T *outPtr = out.get();
 
     // Map labels
-    typedef typename std::map<T, LabelNode<T>* > label_map_t;
-    typedef typename label_map_t::iterator label_map_iterator_t;
+    typedef typename std::unique_ptr< LabelNode<T> > UnqLabelPtr;
+    typedef typename std::map<T, UnqLabelPtr > LabelMap;
+    typedef typename LabelMap::iterator LabelMapIterator;
 
-    label_map_t lmap;
+    LabelMap lmap;
 
     // Initial label
     T label = (T)1;
 
-    for (int j = 0; j < (int)in_dims[1]; j++) {
-        for (int i = 0; i < (int)in_dims[0]; i++) {
-            int idx = j * in_dims[0] + i;
-            if (in_ptr[idx] != 0) {
+    for (int j = 0; j < (int)inDims[1]; j++) {
+        for (int i = 0; i < (int)inDims[0]; i++) {
+            int idx = j * inDims[0] + i;
+            if (inPtr[idx] != 0) {
                 std::vector<T> l;
```

Comment:
It would be better if this was a std::array instead of a std::vector. Since std::arrays are allocated on the stack, you don't get the penalty of calling malloc.

I know its not part of this PR. Perhaps in another issue.